components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint/src/main/webapp/basicauth.jsp

```diff
@@ -160,42 +164,40 @@
                 identityMgtEndpointContext = IdentityUtil.getServerURL("/accountrecoveryendpoint", true, true);
             }
 
-            URL url = null;
-            HttpURLConnection httpURLConnection = null;
+            HttpClient client = new HttpClient();
+            if ("https".equals(scheme)) {
```

Comment:
Shall we move this to a util so that we can use this evrerywhere we want to use http client?